### PR TITLE
CI: Add labels to issue and PR and improve output from submission wf

### LINKED ISSUE #480: Add label to issue and PR during submission wf and improve output
### What is the expected enhancement?

 Add labels :
- `submission` label to every submission issue and PR
- `on hold` label to every submission issue and PR who doesn't pass the standard check
- `ready` label to every submission issue and PR who passed the standard check

Improve output message.

### PR DESCRIPTION
**Pull request by mickahell**
### Summary
Add labels to issue and PR from submission wf.
Improve the wf output.

### Details and comments
- [x] `submission` label to every submission issue and PR
- [x] `on hold` label to every submission issue and PR who doesn't pass the standard check
- [x]  remove `on hold` label if the submission passed
- [x] `ready` label to every submission issue and PR who passed the standard check
- [x] remove `ready` label if the submission doesn't passed anymore
- [x] Add a wf summary
- [x] Make the comment in the issue more clean
- [x] Make the comment in the PR more clean
- [x] Tests run
- [x] Clean

---
Closes #480

### CODE REVIEW

**Comment by mickahell** [comment]
@frankharkins ready for your review :)

I also implemented a new comment structured using the GitHub step summary. That's allow to simplify the code and have more clear output :)
Capture d’écran 2023-08-06 à 17 17 20

All the examples :
- issue passed : https://github.com/mickahell/ecosystem/issues/161
- run summary : https://github.com/mickahell/ecosystem/actions/runs/5777274128
- issue failed : https://github.com/mickahell/ecosystem/issues/160
- run summary : https://github.com/mickahell/ecosystem/actions/runs/5777271007

(the 2 errors in the annotations are normal it's because I'm asking to delete labels who aren't on the issue, I added a `continue-on-error: true` in order to ignore them)

**Comment by mickahell** [comment]
@frankharkins seems the CLA licence is blocked because I added code from GitHub actions. 
I clean this code in my last commit (https://github.com/qiskit-community/ecosystem/pull/481/commits/9a72179f0ff1af8b8c573d6b93819ce491888556), so they aren't any code from GitHub actions anymore but the CLA seems to still not getting it.

**Comment by mickahell** [comment]
The long line of commits was because I had to amend a commit from GitHub action ^^'

File: .github/workflows/ecosystem-submission.yml [hunk 1]
```diff
@@ -125,63 +100,167 @@ jobs:
...
+      uses: actions/github-script@v6
+      if: ${{ steps.check-return.outputs.PASS_STD != 'True' }}
+      with:
+        script: |
+          github.rest.issues.removeLabel({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: ["ready"]
+          })
```
Comment (frankharkins) [inline .github/workflows/ecosystem-submission.yml:260]: Is this duplicate of 194-205?

File: .github/workflows/ecosystem-submission.yml [hunk 2]
```diff
@@ -125,63 +100,167 @@ jobs:
...
 
-    - name: Create comment on failure for dev check
-      if: ${{ steps.check-return.outputs.PASS_DEV != 'True' }}
+    - name: Issue create comment
```
Comment (frankharkins) [inline .github/workflows/ecosystem-submission.yml:262]:
```suggestion
    - name: 'Issue: Post comment'
```

File: .github/workflows/ecosystem-submission.yml [hunk 3]
```diff
@@ -125,63 +100,167 @@ jobs:
...
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: ["on hold"]
+          })
 
-    - name: Create comment on success for dev check
-      if: ${{ steps.check-return.outputs.PASS_DEV == 'True' }}
-      uses: peter-evans/create-or-update-comment@v3
+    - name: Issue labeling status on hold
```
Comment (frankharkins) [inline .github/workflows/ecosystem-submission.yml:238]:
```suggestion
    - name: 'Issue: Add "on hold" label'
```

**Review by frankharkins — CHANGES_REQUESTED** [review]
Look great, thanks!

Only problem is I think a couple of steps are duplicated. I've also suggested some name / comment changes to make things clearer at a glance.

File: .github/workflows/ecosystem-submission.yml [hunk 4]
```diff
@@ -125,63 +100,167 @@ jobs:
...
+      uses: actions/github-script@v6
+      if: ${{ steps.check-return.outputs.PASS_STD == 'True' }}
+      with:
+        script: |
+          github.rest.issues.removeLabel({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: ["on hold"]
+          })
```
[the remark this answers is not in the document]
Comment (mickahell) [reply .github/workflows/ecosystem-submission.yml:237]: Not here is for the issue, on line 170 is for the PR

File: .github/workflows/ecosystem-submission.yml (hunk 1 above)
Comment (mickahell) [reply .github/workflows/ecosystem-submission.yml:261]: Not here is for the issue, on line 170 is for the PR

**Comment by mickahell** [comment]
> Look great, thanks!
> 
> Only problem is I think a couple of steps are duplicated. I've also suggested some name / comment changes to make things clearer at a glance.

Yep @frankharkins I took your feedbacks :)
The duplicated aren't, they are about PR or issue the blocks are almost the same except the variable for the _issue_ id

**Review by frankharkins — CHANGES_REQUESTED** [review]
Sorry to request more changes. I now understand there's a separate step for each label addition/removal; it might be clearer to combine these eight steps into two steps.

That is, this pattern

```
if condition:
    do x
    do y
    do z
```

rather than this one

```
if condition
    do x
if condition
    do y
if condition
    do z
```

Then there should be less code and it should be harder to miss something when modifying the action.

**Comment by mickahell** [comment]
Hmm ok I see what you mean, I'm not sure I can merge stps together. I'l check that

**Comment by frankharkins** [comment]
Can you just put all the code in the same script? E.g.

```yaml
with:
  script: |
    github.rest.issues.addLabel({
      ...
    })
    github.rest.issues.removeLabel({
      ...
    })
```

**Comment by mickahell** [comment]
> Can you just put all the code in the same script? E.g.

Totally ! I was somewhere else and totally forget I can merge script together ^^'

> Sorry to request more changes. I now understand there's a separate step for each label addition/removal; it might be clearer to combine these eight steps into two steps.

Don't worry that's totally ok :)

Done, I mixed them. You can review it

**Review by frankharkins — APPROVED** [review]
Looks good, thank you!

**Pull request merged**